bin/ch/ch.cpp

```diff
@@ -196,48 +207,536 @@ static void CALLBACK PromiseContinuationCallback(JsValueRef task, void *callback
     MessageQueue * messageQueue = (MessageQueue *)callbackState;
 
     WScriptJsrt::CallbackMessage *msg = new WScriptJsrt::CallbackMessage(0, task);
+
+#if ENABLE_TTD
+    ChakraRTInterface::JsTTDNotifyHostCallbackCreatedOrCanceled(true, false, false, task, msg->GetId());
+#endif
+
     messageQueue->InsertSorted(msg);
 }
 
-HRESULT RunScript(const char* fileName, LPCWSTR fileContents, BYTE *bcBuffer, char *fullPath)
+void StartupDebuggerAsNeeded()
 {
-    HRESULT hr = S_OK;
-    MessageQueue * messageQueue = new MessageQueue();
-    WScriptJsrt::AddMessageQueue(messageQueue);
-    LPWSTR fullPathWide = nullptr;
+    if(dbgIPAddr == nullptr)
+    {
+        if(doTTDebug)
+        {
+            //we need to force the script context into dbg mode for replay even if we don't attach the debugger -- so do that here
+            ChakraRTInterface::JsTTDSetDebuggerForReplay();
+        }
+    }
+    else
+    {
+        //
+        //TODO: not sure how to set IP and other stuff here !!!
+        //
+        AssertMsg(false, "Broken");
 
-    IfJsErrorFailLog(ChakraRTInterface::JsSetPromiseContinuationCallback(PromiseContinuationCallback, (void*)messageQueue));
+        Debugger* debugger = Debugger::GetDebugger(chRuntime);
+        debugger->StartDebugging(chRuntime);
+    }
+}
+
+void CreateDirectoryIfNeeded(const char16* path)
```

Comment:
Please move helper functions which are generic to Helpers.cpp like CreateDirectoryIfNeeded, DeleteDirectory, GetFileFromURI